src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts

```diff
@@ -371,6 +371,33 @@ export class BadgeClassEditFormComponent extends BaseAuthenticatedRoutableCompon
 		this.fetchTags();
 	}
 
+	async handleBadgeCategoryChange() {
+		if (
+			this.badgeCategory === 'competency' &&
+			this.badgeClassForm.rawControl.controls['badge_category'].value !== 'competency'
+		) {
+			if (
+				await this.dialogService.confirmDialog.openTrueFalseDialog({
+					dialogTitle: 'Wenn du die Kategorie änderst, werden alle Kompetenzen gelöscht.',
+					dialogBody: 'Möchtest du fortfahren?',
+					resolveButtonLabel: 'Fortfahren',
+					rejectButtonLabel: 'Abbrechen',
+				})
+			) {
+				this.badgeClassForm.controls.competencies.reset();
+				const controls = this.badgeClassForm.controls.competencies.controls;
+				for (let i = controls.length - 1; i >= 0; i--) {
+					this.badgeClassForm.controls.competencies.removeAt(controls.indexOf(controls[i]));
```

Comment:
Why is this so complicated? Like, why do you delete the competencies at the index of the controls of the competencies? Can't you just directly delete all competencies, e.g. with a `clear` call or sth similar? And if not, why don't iterate over the competencies instead of the controls of the competencies?

Reply:
My bad, iterating over competencies now